### PR TITLE
add support for nested attribute hashes rendered as hyphen separated flat attributes

### LINKED ISSUE #450: hyphenated tag attributes are poorly supported, making stimulus integration painful
I'm suprised as they seem to work correctly in ActiveAdmin forms, but in Arbre, hyphenated tag attributes are difficult to emit.

In an ActiveAdmin form, I can do the following:
      `f.input(:trade, input_html: { data: { action: 'click->trade#foo' } })`
The expected hyphenated `data-action="click->trade#foo'` attribute will come out in the HTML, which works seamlessly with stimulus.

If I'm rendering an Arbre partial template, the same does not yield the desired results:

In order to get hyphenated attributes, I ended up doing this:
`div(class: 'myclass',  **{ 'data-controller'.to_sym => 'worker-cred' }) do`
Passing a straight hash as the second parameter works for some tags, but didn't work for a div with a block.  I didn't dig into the why, but using `to_sym` and splatting the hash eventually got me the desired result.

That code ends up appropriately rendering a data-controller attribute for the div (and working with stimulus), but it's definitely not intuitive to anyone who's not pretty deep in ruby magic.

If instead I put the following in the Arbre template:
`div(class: 'myclass',  data: { controller: 'worker-cred' }) do`

I end up with html like this: 
`<div class="myclass" data="{:controller=>&quot;worker-cred&quot;}">`

I believe the code  is in `lib/arbre/html/attributes.rb`  in the `to_s` method .    What are the chances of updating this code to allow the form that works in .erb and ActiveAdmin forms, in order to better integrate with Stimulus?

It looks like the ActionView code for rendering tag attributes can be found here:
[/home/rbecker/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/actionview-7.0.2.4/lib/action_view/helpers/tag_helper.rb](https://github.com/rails/rails/blob/e184d33dfcef9e7265781ec84af559af4781765c/actionview/lib/action_view/helpers/tag_helper.rb#L89)

### PR DESCRIPTION
**Pull request by Ikariusrb**
Closes #450

### CODE REVIEW

**Review by javierjulio — COMMENTED** [review]
This looks great, thanks! Only a minor note on the context grouping since I figured what's here was unintentional. 

Would you mind adding another context with the same test case pairing but testing that a data attribute as a full named key (with hyphens, not using nested hashes) works as expected?

File: spec/arbre/unit/html/tag_attributes_spec.rb [hunk 1]
```diff
@@ -13,16 +13,50 @@
...
+          tag.class_list # initializes an empty ClassList
+          tag.set_attribute :foo, { bar: '' }
+          tag.set_attribute :bar, { baz: nil }
+
+          expect(tag.to_s).to eq "<tag id=\"my_id\" data-action=\"some_action\"></tag>\n"
+        end
+      end
+    end
+
+    context "when there is a deeply nested attribute" do
```
[the remark this answers is not in the document]
Comment (Ikariusrb) [reply spec/arbre/unit/html/tag_attributes_spec.rb]: Whoops, you're absolutely correct.  Done!

**Comment by Ikariusrb** [comment]
> Would you mind adding another context with the same test case pairing but testing that a data attribute as a full named key (with hyphens, not using nested hashes) works as expected?

I'm not quite certain what you're asking for here.  You're asking for a test showing that the (exceedingly awkward) code that I (ab)used to get Arbre to emit a hyphenated tag before still works?  If that's what you're asking for, I can add a test easily, but IMO the code I used to do that wass terrible and should never be encouraged, and adding a test to codify it as part of the developer contract would be encouragement in my eyes.

If that's not what you're asking for, I'm a bit lost, and require further clarification :)

**Comment by javierjulio** [comment]
@Ikariusrb no but I think we should still address the original issue. I'm good with the support here for nested hashes, so thank you! My understanding is that with your changes, it should resolve the original issue you had right? For example, this test case passes with your changes.

```ruby
      it "should render the attributes to html" do
        tag.build id: "my_id", "data-action" => "action"
        expect(tag.to_s).to eq "<tag id=\"my_id\" data-action=\"action\"></tag>\n"
      end
```

I was thinking for completion it would be nice to have a set of tests like this in the pattern you set to demonstrate that hyphenated keys rather than nested hashes also works as expected. What do you think?

File: spec/arbre/unit/html/tag_attributes_spec.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (javierjulio) [reply spec/arbre/unit/html/tag_attributes_spec.rb]: Thanks!

**Comment by Ikariusrb** [comment]
My PR doesn't change the ability to pass in hyphenated attributes directly at all.  The problem is ruby itself doesn't support hyphenated symbols unless you construct them via `.to_sym`,  and Arbre expects to receive the attributes as keyword args, so you have to construct a hash with rocket to get a hyphenated key in, then double-splat it out to the keyword args to get it passed in as Arbre expects.  Changing the call signature for the tag builders is something I'd rather not attempt to touch, as that's far more likely to break something than what I added.

**Comment by Ikariusrb** [comment]
@javierjulio - any update on the ability for this to be merged?

**Comment by javierjulio** [comment]
@Ikariusrb yes, I will get to this soon, thank you.

**Comment by javierjulio** [comment]
@Ikariusrb I rebased to get CI green again on the latest. I reviewed this and realize why this makes no changes to hyphenated attributes. I remember now why I asked though since I wanted to have a test case to demonstrate that it works as we didn't have a test for that. The reason why is that [the arguments are forwarded and the last arg is a hash](https://github.com/activeadmin/arbre/blob/b09b8131afc841459c79397216bb5d5cac97b06f/lib/arbre/html/tag.rb#L19-L40), so keyword arguments in this case should not matter. It wasn't dependent on your change. It's a related request to improve the tests.

**Review by javierjulio — APPROVED** [review]
Thanks!

**Pull request merged**